tracer/tracer.go

```diff
@@ -23,6 +34,13 @@ type Tracer struct {
 
 	DebugLoggingEnabled bool
 	enabled             bool // defines if the Tracer is enabled or not
+
+	services         map[string]*Service // name -> service
+	servicesModified bool
+	serviceChan      chan *Service
```

Comment:
Here not sure the level of pointer indirection is worth it, any particular reason except tiny perf boost?